cli-plugins/hooks/template.go

```diff
@@ -0,0 +1,117 @@
+package hooks
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+type HookType int
+
+const (
+	NextSteps = iota
+)
+
+// HookMessage represents a plugin hook response. Plugins
+// declaring support for CLI hooks need to print a json
+// representation of this type when their hook subcommand
+// is invoked.
+type HookMessage struct {
+	Type     HookType
+	Template string
+}
```

Comment:
Do we expect a single invocation of the hook to return a single message, or could there be situations where we want multiple to be returned? (And in that case, should we have some struct with a slice of HookMessages to be returned?

Reply:
Just to say it's quite common inside docker scout cli to display multiple hooks from a single command. It's different as it's not in the main CLI, but just to share the idea that multiple hooks can be interesting at some point.